### PR TITLE
Fix section naming in docstring causing visualization problems

### PR DESCRIPTION
**Pull request by morcuended**
**Description**

This pull request fixes some docstring formatting issues that were causing warnings in docs deployment and some docs visualization problems (these sections were not being shown).

Also fixes a broken link to GADF docs related to the observation modes section.

Also improves the readability of a paragraph in [spectral calculation with energy-dependent directional cuts](https://docs.gammapy.org/1.1/tutorials/analysis-1d/spectral_analysis_rad_max.html).

Solves #4760 

**Dear reviewer**

### CODE REVIEW

File: examples/tutorials/analysis-1d/spectral_analysis_rad_max.py [hunk 1]
```diff
@@ -41,15 +41,15 @@
...
-an energy- (and offset-) dependent :math:`\\theta` cut is applied, its
-values are stored in additional `FITS` data unit, named
-``RAD_MAX_2D` <https://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/point_like/#rad-max-2d>`__.
+specifications offer two different ways to store this information:
+
+* if the same :math:`\\theta` cut is applied at all energies and offsets, a
+  `RAD_MAX <https://gamma-astro-data-formats.readthedocs.io/en/latest/irfs/point_like/#rad-max>`__
+  keyword is added to the header of the data units containing IRF components. This
+  should be used to define the size of the ON and OFF regions;
+* in case an energy- (and offset-) dependent :math:`\\theta` cut is applied, its
```
Comment (Astro-Kirsty) [inline examples/tutorials/analysis-1d/spectral_analysis_rad_max.py:50]: I would adjust this too:
`energy-dependent (and offset-dependent) `

**Review by Astro-Kirsty — CHANGES_REQUESTED** [review]
Thanks a lot @morcuended 

All good on my side.
Just one small comment

File: examples/tutorials/analysis-1d/spectral_analysis_rad_max.py (hunk 1 above)
Comment (morcuended) [reply examples/tutorials/analysis-1d/spectral_analysis_rad_max.py]: done

**Review by registerrier — APPROVED** [review]
Thanks @morcuended . This looks good!

**Comment by registerrier** [comment]
CI fail is unrelated.

**Pull request merged**